chalicelib/modules/mongo.py

```diff
@@ -159,21 +159,30 @@ def update_document_by_id(
             return False
 
     @add_env_suffix
-    def update_document(self, collection, document_id, query):
+    def update_document(self, collection, document_id, query, array_filters=None):
         """
         Updates a document in the specified collection with the given ID.
 
         Args:
             collection (str): The name of the collection to update the document in.
             document_id (str): The ID of the document to update.
             query (dict): A dictionary containing the update operators.
+            array_filters (list, optional): A list of filters to apply when updating 
+                                            elements in an array field of the document.
+                                            Each filter in the list is a dictionary 
+                                            specifying the criteria for selecting 
+                                            array elements to be updated. Default is None.
 
         Returns:
             bool: True if the update was successful, False otherwise.
         """
         try:
+            update_options = {}
```

Comment:
Not too sure how this works, but would there be a way to guarantee that right filters get passed in?

Reply:
Ya lol, this was ChatGPT because updating documents with arrays of objects requires this `array_filters` thing (basically queries the array for the matching `elementId` and performs whatever operation is supplied).

I'll add some type-validation to ensure that array_filters is an appropriate list of dictionaries.